lib/AST/Type.cpp

```diff
@@ -673,8 +673,11 @@ ClassDecl *CanType::getClassBoundImpl(CanType type) {
   llvm_unreachable("class has no class bound!");
 }
 
-LayoutConstraintInfo CanType::getLayoutConstraintInfo() const {
-  return swift::getLayoutConstraintInfo(*this);
+LayoutConstraint CanType::getLayoutConstraint() const {
```

Comment:
There's no need to canonicalize the type first -- just do if (auto archetypeTy = getAs<ArchetypeType>(this)) { ... }